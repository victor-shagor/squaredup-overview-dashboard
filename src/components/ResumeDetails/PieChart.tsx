import React from "react";

import Chart from "react-apexcharts";
import { PieChartProps } from "constants/interfaces";

const PieChart = ({ label, colors, series }: PieChartProps) => {
  const options = {
    labels: label,
    colors: colors,
    dataLabels: {
      enabled: false,
    },

    plotOptions: {
      pie: {
        donut: {
          size: "58%",
          labels: {
            show: true,
            name: {
              show: true,
              fontSize: "22px",
              fontFamily: "Helvetica, Arial, sans-serif",
              fontWeight: 600,
              color: undefined,
              offsetY: 10,
            },
            value: {
              show: true,
              fontSize: "16px",
              fontWeight: "bold",
              color: undefined,
              offsetY: -20,
            },
            total: {
              show: true,
              showAlways: true,
              label: "Total Resumes",
              fontSize: "15px",
              fontFamily: "SFProText",
              color: "#373d3f",
            },
          },
        },
      },
    },
  };
  return (
    <div>
      <Chart
        options={{
          legend: {
            position: "right",
            offsetY: 100,
            height: 300,
          },
          ...options,
        }}
        type="donut"
        width={500}
        height={320}
        series={series}
      />
    </div>
  );
};

export default PieChart;
