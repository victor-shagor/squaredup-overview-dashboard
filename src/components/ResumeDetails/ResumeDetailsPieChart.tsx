import { ResumeDetailsPieChartProps } from "constants/interfaces";
import PieChart from "components/PieChart";

const ResumeDetailsPieChart = ({
  label,
  series,
}: ResumeDetailsPieChartProps) => {
  const options = {
    labels: label,
    colors: ["#5a5bb5", "#5dace6"],
    dataLabels: {
      enabled: false,
    },

    plotOptions: {
      pie: {
        donut: {
          size: "58%",
          labels: {
            show: true,
            name: {
              show: true,
              fontSize: "22px",
              fontFamily: "Helvetica, Arial, sans-serif",
              fontWeight: 600,
              color: undefined,
              offsetY: 10,
            },
            value: {
              show: true,
              fontSize: "16px",
              fontWeight: "bold",
              color: undefined,
              offsetY: -20,
            },
            total: {
              show: true,
              showAlways: true,
              label: "Total Resumes",
              fontSize: "15px",
              fontFamily: "SFProText",
              color: "#373d3f",
            },
          },
        },
      },
    },
  };
  return (
    <div>
      <PieChart options={options} width={500} height={320} series={series} />
    </div>
  );
};

export default ResumeDetailsPieChart;
