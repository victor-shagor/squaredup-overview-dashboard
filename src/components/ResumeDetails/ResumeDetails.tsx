import { ResumeDetailsProps } from "constants/interfaces";
import { Card } from "pages/overview/overview.styles";
import ResumeDetailsPieChart from "./ResumeDetailsPieChart";

const ResumeDetails = ({ sharedAndUnsharedResumes }: ResumeDetailsProps) => {
  return (
    <Card
      height={"25rem"}
      width={"46.4%"}
      mobile={false}
      data-testid="resume-details-container"
    >
      <div className="details">
        <p>Resume Details</p>
      </div>
      <div className="piechart">
        <ResumeDetailsPieChart
          label={["Shared Resume", "Unshared Resume"]}
          series={[
            sharedAndUnsharedResumes.sharedResumes,
            sharedAndUnsharedResumes.unSharedResume,
          ]}
        />
      </div>
    </Card>
  );
};

export default ResumeDetails;
