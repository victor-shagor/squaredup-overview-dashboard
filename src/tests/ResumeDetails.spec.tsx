import { render, screen } from "@testing-library/react";
import ResumeDetails from "components/ResumeDetails/ResumeDetails";
import renderer from "react-test-renderer";

jest.mock("react-apexcharts", () => {
  return {
    __esModule: true,
    default: () => {
      return <div />;
    },
  };
});

const mockData = {
  sharedResumes: 10,
  unSharedResume: 90,
};

describe("resume details component", () => {
  it("render active user component with the correct style", () => {
    render(<ResumeDetails sharedAndUnsharedResumes={mockData} />);
    expect(screen.getByTestId("resume-details-container")).toHaveStyle(
      `height:25rem`
    );
    expect(screen.getByTestId("resume-details-container")).toHaveStyle(
      `width:46.4%`
    );
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(<ResumeDetails sharedAndUnsharedResumes={mockData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
