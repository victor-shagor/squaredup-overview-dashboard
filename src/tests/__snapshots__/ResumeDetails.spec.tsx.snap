// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resume details component renders correctly 1`] = `
<div
  className="sc-gsnTZi kJQinY"
  data-testid="resume-details-container"
  height="25rem"
  width="46.4%"
>
  <div
    className="details"
  >
    <p>
      Resume Details
    </p>
  </div>
  <div
    className="piechart"
  >
    <div>
      <div
        data-testid="piechart-container"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
